### PR TITLE
Add .filter to UseListStateHandlers

### PR DESCRIPTION
**Pull request by alimehasin**
This PR adds .filter to UseListStateHandlers

### CODE REVIEW

File: docs/src/docs/hooks/use-list-state.mdx [hunk 1]
```diff
@@ -63,6 +63,10 @@ const setItem = () => handlers.setItem(0, { a: 8 });
 // set item property at given position
 const setItemProp = () => handlers.setItemProp(1, 'a', 'new-prop');
 // values -> [{ a: 8 }, { a: 'new-prop' }]
+
+// remove objects that have a = 'new-props'
+const filter = () => handlers.filter((item) => item.a === 'new-props');
```
Comment (rtivital) [inline docs/src/docs/hooks/use-list-state.mdx:68]: Invalid operation, item value is `new-prop`, not `new-props`

File: docs/src/docs/hooks/use-list-state.mdx [hunk 2]
```diff
@@ -63,6 +63,10 @@ const setItem = () => handlers.setItem(0, { a: 8 });
...
+// values -> [{ a: 8}]
```
Comment (rtivital) [inline docs/src/docs/hooks/use-list-state.mdx:69]: Incorrect code formatting, should be `[{ a: 8 }]` instead

File: docs/src/docs/hooks/use-list-state.mdx [hunk 3]
```diff
@@ -83,6 +87,7 @@ Hook provides handlers to work with array data:
 - `setItem` – replace item at given index
 - `setItemProp` – set item property at given index
 - `setState` – set list state with react action
+- `filter` - filter your state just like you filter an array
```
Comment (rtivital) [inline docs/src/docs/hooks/use-list-state.mdx:90]: Change to "`filter` - filter values with callback function"

File: src/mantine-hooks/src/use-list-state/use-list-state.ts [hunk 4]
```diff
@@ -16,6 +16,7 @@ export interface UseListStateHandlers<T> {
   reorder: ({ from, to }: { from: number; to: number }) => void;
   setItem: (index: number, item: T) => void;
   setItemProp: <K extends keyof T, U extends T[K]>(index: number, prop: K, value: U) => void;
+  filter: (fn: (item: T) => boolean) => void;
```
Comment (rtivital) [inline src/mantine-hooks/src/use-list-state/use-list-state.ts:19]: `fn` should have an index as second argument

File: src/mantine-hooks/src/use-list-state/use-list-state.ts [hunk 5]
```diff
@@ -82,6 +83,13 @@ export function useListState<T>(initialValue: T[] = []): UseListState<T> {
       current.map((item, index) => (condition(item, index) ? fn(item, index) : item))
     );
 
+  const filter = (fn: (item: T) => boolean) => {
+    setState((current) => {
+      const cloned = [...current];
```
Comment (rtivital) [inline src/mantine-hooks/src/use-list-state/use-list-state.ts:88]: This operation is redundant, filter method already creates a shallow copy of the array, remove it

**Review by rtivital — CHANGES_REQUESTED** [review]

**Comment by rtivital** [comment]
Thanks!

**Comment by alimehasin** [comment]
my pleasure

**Pull request merged**